src/PartitionGraph.f90

```diff
@@ -2209,4 +2229,187 @@ SUBROUTINE swapDomain(thisGraph,iv,linL1,D,lswap)
       IF(lswap) linL1(iv)=(.NOT. linL1(iv))
     ENDSUBROUTINE swapDomain
 !
+!-------------------------------------------------------------------------------
+!> @brief A fixup for discontiguous domains during recursive partitioning
+!> @param thisGraph The graph object
+!> @param Order The order vector
+!> @param nv1 The number of vertices in sub-domain 1
+!> @param lContiguous Logical for if the two subdomains are contiguous
+!>
```

Comment:
Provide a brief high level description of how this routine takes to domains, checks for contiguity, and fix ups the contiguousness of the domains.